jocd/src/br/org/certi/jocd/board/MbedBoard.java

```diff
@@ -0,0 +1,281 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the license for the specific language governing permissions and
+ * limitations under the license.
+ */
+/*
+ *###################################################################
+ *# Copyright (C) 2018 Fundação CERTI
+ *# All Rights Reserved
+ *###################################################################
+ */
+package br.org.certi.jocd.board;
+
+import android.content.Context;
+import android.os.SystemClock;
+import android.text.TextUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import br.org.certi.jocd.DAPAccess.DAPAccessCMSISDAP;
+import br.org.certi.jocd.target.TargetFactory;
+import br.org.certi.jocd.target.TargetFactory.targetEnum;
+
+public class MbedBoard extends Board {
+
+    // Logging
+    private static final String TAG = "MbedBoard";
+
+    public final String uniqueId;
+    public final String boardId;
+    public final String name;
+    public String testBinary;
+
+    /*
+     * Constructor.
+     */
+    public MbedBoard(DAPAccessCMSISDAP link, TargetFactory.targetEnum targetOverride, int frequency) throws UnsupportedBoardException {
+        super();
+
+        this.uniqueId = link.getUniqueId();
+        boardId = this.uniqueId.substring(0,4);
+
+        TargetFactory.targetEnum target = null;
+        BoardInfo boardInfo = BoardInfo.getBoardInfo(boardId);
+        if (boardInfo != null) {
+            // Get the board name.
+            this.name = boardInfo.name;
+            // Get the native target.
+            target = boardInfo.target;
+            // Get the binary test.
+            this.testBinary = boardInfo.binary;
+        }
+        else {
+            this.name = "Unknown Board";
+        }
+
+        // Override the native target... (if supposed to)
+        if (targetOverride != null) target = targetOverride;
+
+        // Check if we have a valid target.
+        if (target == null) {
+            throw new UnsupportedBoardException(boardId);
+            // TODO
+            // pyOCD uses cotex_m here. Check if it makes sense.
+            // (mved_board.py line 130)
+        }
+
+
+        super.setup(link, target, frequency);
+    }
+
+    /*
+     * Overload for getAllConnectedBoards using default values.
+     */
+    public static List<MbedBoard> getAllConnectedBoards(Context context) {
+        return getAllConnectedBoards(context, false, true, null, 0);
+    }
+
+    /*
+     * Return an array of all mbed boards connected.
+     */
+    public static List<MbedBoard> getAllConnectedBoards(Context context, boolean close,
+                                                        boolean blocking, targetEnum targetOverride,
+                                                        int frequency) {
+        List<MbedBoard> mbedList = new ArrayList<MbedBoard>();
+
+        boolean unsupportedBoard = false;
```

Comment:
Does this boolean have any intended future use?

Reply:
No! It wasn't supposed to be here. (Fixed!)